capsules/extra/src/can.rs

```diff
@@ -292,7 +292,7 @@ impl<'a, Can: can::Can> SyscallDriver for CanCapsule<'a, Can> {
             7 => {
                 self.can_rx
                     .take()
-                    .map(|dest_buffer| {
+                    .map_or(CommandReturn::failure(ErrorCode::NOMEM), |dest_buffer| {
```

Comment:
Not that I think there's anything wrong with this, but what triggered this change?

Reply:
Clippy, trying to make our code more idiomatic. This was a hard-error in CI. I'm not neither happy about nor opposed to it complaining here.